referenceframe/frame_system.go

```diff
@@ -79,12 +80,77 @@ type simpleFrameSystem struct {
 	parents map[Frame]Frame
 }
 
-// NewEmptySimpleFrameSystem creates a graph of Frames that have.
-func NewEmptySimpleFrameSystem(name string) FrameSystem {
+// NewEmptyFrameSystem creates a graph of Frames that have.
+func NewEmptyFrameSystem(name string) FrameSystem {
 	worldFrame := NewZeroStaticFrame(World)
 	return &simpleFrameSystem{name, worldFrame, map[string]Frame{}, map[Frame]Frame{}}
 }
 
+// NewFrameSystem assembles a frame system from a set of parts and additional transforms.
+func NewFrameSystem(name string, parts []*FrameSystemPart, additionalTransforms []*LinkInFrame) (FrameSystem, error) {
+	allParts := make([]*FrameSystemPart, 0, len(parts)+len(additionalTransforms))
+	allParts = append(allParts, parts...)
+	for _, tf := range additionalTransforms {
+		transformPart, err := LinkInFrameToFrameSystemPart(tf)
+		if err != nil {
+			return nil, err
+		}
+		allParts = append(allParts, transformPart)
+	}
+
+	// ensure that at least one frame connects to world if the frame system is not empty
+	if len(allParts) != 0 {
+		hasWorld := false
+		for _, part := range allParts {
+			if part.FrameConfig.Parent() == World {
+				hasWorld = true
+				break
+			}
+		}
+		if !hasWorld {
+			return nil, ErrNoWorldConnection
+		}
+	}
+	// Topologically sort parts
+	sortedParts, err := TopologicallySortParts(allParts)
+	if err != nil {
+		return nil, err
+	}
+	if len(sortedParts) != len(allParts) {
+		return nil, errors.Errorf(
+			"frame system has disconnected frames. connected frames: %v, all frames: %v",
+			getPartNames(sortedParts),
+			getPartNames(allParts),
+		)
+	}
+	fs := NewEmptyFrameSystem(name)
+	for _, part := range sortedParts {
+		// rename everything with prefixes
+		part.FrameConfig.SetName(part.FrameConfig.Name())
```

Comment:
It's not at all clear why calling `SetName()` with what's returned by `Name()` on the same object will result in anything at all different; could you clarify (and put in a comment in the code)?

Reply:
To leave a paper trail from an offline conversation: this used to use a prefix but this is no longer needed since the prefix was never used anywhere.  Therefore this whole chunk of code can be removed!